nemo_aligner/utils/trt_llm.py

```diff
@@ -0,0 +1,175 @@
+import secrets
+
+import tensorrt_llm
+import torch
+
+from nemo.export.tensorrt_llm import TensorRTLLM
+from nemo.export.trt_llm import tensorrt_llm_run
+from nemo.export.trt_llm.nemo_ckpt_loader.nemo_file import build_tokenizer
+from nemo.utils import logging
+from nemo_aligner.utils import parallel_state
+from nemo_aligner.utils.distributed import broadcast_2d_tensor_within_mp
+
+
+class GPTGenerateTRTLLM:
+    def __init__(
+        self,
+        model_cfg,
+        max_generation_length=1024,
+        max_input_len=1024,
+        max_input_tokens=4096,
+        generation_batch_size=4,
+        unload_engine_train=False,
+        trt_model_type="GPTForCausalLM",
+        end_strings=None,
+        reshard_model=False,
+        sample_temperature=None,
+        sample_top_k=None,
+        sample_top_p=None,
+        repetition_penalty=1.0,
+        use_greedy=False,
+        tokenizer=None,
+        trt_model_dir="/tmp/trt_llm_model",
+        seed=None,
+    ):
+        if use_greedy and sample_top_k != 1:
+            if sample_top_k != 1:
```

Comment:
This second `if` seems useless given the first one.

Reply:
6c2d4581ea83507ac1775940df5743b41c9e90e5